HopsanGUI/CoreAccess.cpp

```diff
@@ -475,6 +476,8 @@ bool CoreSystemAccess::setParameterValue(QString componentName, QString paramete
 {
     hopsan::Component *pComponent = mpCoreComponentSystem->getSubComponent(componentName.toStdString().c_str());
     bool retval = pComponent->setParameterValue(parameterName.toStdString().c_str(), value.toStdString().c_str(), force);
+    pComponent->reconfigure();
```

Comment:
Actually I want it to be called also when loading. If you load a model with an existing FMI component, it should reconfigure the component properly to make sure all ports and parameters exist. We could add some information to the CAF file so that only certain parameters calls reconfigure, but won't that introduce even more overhead? Calling an empty function is not so time consuming. And the components can have an internal check to make sure they only perform a reconfiguration when a certain critical parameter has changed.

Reply:
"Tagging" a parameter must be in the core, so that it can reconfigure when run from CLI or as an exported FMU or similar, then CAF files are not used.

In case a component have reconfigure logic, it will run every time you change a parameter, or in this case, every time you close a properties dialog, once for each parameter and once for each start value. (Unless the dialog box ignores values not changed, but I do not think so).

Currently reconfigure is only handled by the GUI, it would be better if the core could decide when to call it maybe. Then applications would not need extra logic to decide.

In the load case, it could be called once for each component after load is complete.

Reply:
I added an  "is reconfigurable" flag to `Parameter` class, with an `addReconfigurationParameter()` method in the component API. This way, each component can notify the core that a parameter triggers a reconfiguration. All reconfiguration except graphical appearance is then handled by the core.

Perhaps the name of the function should be changed, as it does not actually add the parameter. You need to first add it using `addConstant/InputVariable` and then call the new function as well to make it reconfigurable. I did not want to bloat the existing add functions with a new boolean parameter.